### PR TITLE
snippets for sweeping funds on channel closes.

### PR DESCRIPTION
**Pull request by ubbabeck**

### CODE REVIEW

File: snippets/go/closed_channel.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	"log"
+
+	"github.com/breez/breez-sdk-go/breez_sdk"
+)
+
+func PrepareSweep(feeRate uint64) {
+	// ANCHOR: prepare-sweep
+	satPerVbyte := feeRate
+	destinationAddress := "bc1.."
+	req, err := sdk.PrepareSweep(breez_sdk.PrepareSweepRequest{SatPerVbyte: satPerVbyte, ToAdrees: &destinationAddress})
```
Comment (dangeross) [inline snippets/go/closed_channel.go:13]:
```suggestion
	req, err := sdk.PrepareSweep(breez_sdk.PrepareSweepRequest{SatPerVbyte: satPerVbyte, ToAddress: destinationAddress})
```

File: snippets/go/closed_channel.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+	if err != nil {
+		log.Fatalf("PrepareSweep error %#v", err)
```
Comment (dangeross) [inline snippets/go/closed_channel.go:15]: Just to be consistent with other examples
```suggestion
		log.Printf("PrepareSweep error %#v", err)
```

File: snippets/go/closed_channel.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	}
+	// ANCHOR_END: prepare-sweep
+}
+
+func Sweep(satPerVbyte uint32, toAddress string) {
+	// ANCHOR: sweep
+	req, err := sdk.Sweep(breez_sdk.SweepRequest{SatPerVbyte: satPerVbyte, ToAdrees: &toAddress})
```
Comment (dangeross) [inline snippets/go/closed_channel.go:22]:
```suggestion
	req, err := sdk.Sweep(breez_sdk.SweepRequest{SatPerVbyte: satPerVbyte, ToAddress: toAddress})
```

File: snippets/go/closed_channel.go [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+	if err != nil {
+		log.Fatalf("Sweep error %#v", err)
```
Comment (dangeross) [inline snippets/go/closed_channel.go:24]:
```suggestion
		log.Printf("Sweep error %#v", err)
```

File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ClosedChannel.kt [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+import breez_sdk.*
+
+class ClosedChannel {
+    func prepareSweep(sdk: BlockingBreezServices, feeRate: int) {
+        // ANCHOR: prepare-sweep 
+         val satPerVbyte = feeRate
+         var destinationAddress = "bc1.."
+         try {
+            val req = PrepareSweepReq(satPerVbyte, destinationAddress)
```
Comment (dangeross) [inline snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ClosedChannel.kt:11]:
```suggestion
            val req = PrepareSweepRequest(satPerVbyte, destinationAddress)
```

File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ClosedChannel.kt [hunk 6]
```diff
@@ -0,0 +1,31 @@
+package com.example.kotlinmpplib
+
+import breez_sdk.*
+
+class ClosedChannel {
+    func prepareSweep(sdk: BlockingBreezServices, feeRate: int) {
```
Comment (dangeross) [inline snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ClosedChannel.kt:6]: I think uint64 is ULong
```suggestion
    func prepareSweep(sdk: BlockingBreezServices, feeRate: ULong) {
```

File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ClosedChannel.kt [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+            val req = PrepareSweepReq(satPerVbyte, destinationAddress)
+            prepareSweepResp = sdk.prepareSweep(req)
+         }
+         catch (e: Exception) {
+            // handle error
+        }
+        // ANCHOR_END: prepare-sweep 
+    }
+
+    func sweep(sdk: BlockingBreezServices, satPerVbyte: int, toAddress) {
```
Comment (dangeross) [inline snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ClosedChannel.kt:20]: For some reason satPerVbyte is uint32 here, so UInt
```suggestion
    func sweep(sdk: BlockingBreezServices, satPerVbyte: UInt, toAddress: String) {
```

File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ClosedChannel.kt [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+        // ANCHOR: sweep 
+         try {
+            val req = SweepReq(satPerVbyte, toAddress)
```
Comment (dangeross) [inline snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ClosedChannel.kt:23]:
```suggestion
            val req = SweepRequest(satPerVbyte, toAddress)
```

File: snippets/python/src/closed_channel.py [hunk 9]
```diff
@@ -0,0 +1,20 @@
+import breez_sdk
+def prepare_sweep(fee_rate):
+    # ANCHOR: prepare_sweep
+    destination_address = "bc1.."
+    sat_per_vbyte = fee_rate
+    try:
+        req = breez_sdk.PrepareSweepRequest(fee_rate,  destination_address)
```
Comment (dangeross) [inline snippets/python/src/closed_channel.py:7]: Have to be careful of the ordering in python, it takes the to_address as the first param
```suggestion
        req = breez_sdk.PrepareSweepRequest(destination_address, fee_rate)
```

File: snippets/python/src/closed_channel.py [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+    # ANCHOR_END: prepare_sweep
+    except Exception as error:
+        print(error)
+        raise
+
+def sweep(to_address, fee_rate):
+    try:
+        # ANCHOR: sweep
+        req = breez_sdk.SweepRequest(fee_rate, to_address)
```
Comment (dangeross) [inline snippets/python/src/closed_channel.py:16]: Same here
```suggestion
        req = breez_sdk.SweepRequest(to_address, fee_rate)
```

File: snippets/react-native/closed_channel.ts [hunk 11]
```diff
@@ -0,0 +1,20 @@
+import {
+    type PrepareSweep,
+    prepareSweep,
+    p
+  } from '@breeztech/react-native-breez-sdk'
+
+  const prepareSweep =async (feeRate: BigInt) => {
```
Comment (dangeross) [inline snippets/react-native/closed_channel.ts:7]:
```suggestion
  const prepareSweep = async (feeRate: number) => {
```

File: snippets/react-native/closed_channel.ts [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+    // ANCHOR: prepare-sweep
+    const destinationAddress = "bc1.."
+    const satPerVbyte = feeRate
+
+    const prepareSweepResp = await prepareSweep({ destinationAddress, satPerVbyte })
```
Comment (dangeross) [inline snippets/react-native/closed_channel.ts:12]: Variable has to match the type field, or use `{ toAddress: destinationAddress, ... }`
```suggestion
    const toAddress = "bc1.."
    const satPerVbyte = feeRate

    const prepareSweepResp = await prepareSweep({ toAddress, satPerVbyte })
```

File: snippets/react-native/closed_channel.ts [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+    // ANCHOR_END: prepare-sweep
+  }
+
+  const sweep =async (satPerVbyte: BigInt, toaddress: String) => {
+    // ANCHOR: sweep
+    const prepareSweepResp = await sweep({ toaddress, satPerVbyte })
```
Comment (dangeross) [inline snippets/react-native/closed_channel.ts:18]:
```suggestion
  const sweep = async (satPerVbyte: number, toAddress: string) => {
    // ANCHOR: sweep
    const prepareSweepResp = await sweep({ toAddress, satPerVbyte })
```

File: snippets/react-native/closed_channel.ts [hunk 14]
```diff
@@ -0,0 +1,20 @@
+import {
+    type PrepareSweep,
+    prepareSweep,
+    p
```
Comment (dangeross) [inline snippets/react-native/closed_channel.ts:4]:
```suggestion
    prepareSweep,
    sweep
```

File: snippets/python/src/closed_channel.py [hunk 15]
```diff
@@ -0,0 +1,20 @@
+import breez_sdk
+def prepare_sweep(fee_rate):
+    # ANCHOR: prepare_sweep
```
Comment (dangeross) [inline snippets/python/src/closed_channel.py:3]: nit: for consistency
```suggestion
    # ANCHOR: prepare-sweep
```

File: snippets/python/src/closed_channel.py [hunk 16]
```diff
@@ -0,0 +1,20 @@
...
+    destination_address = "bc1.."
+    sat_per_vbyte = fee_rate
+    try:
+        req = breez_sdk.PrepareSweepRequest(fee_rate,  destination_address)
+    # ANCHOR_END: prepare_sweep
```
Comment (dangeross) [inline snippets/python/src/closed_channel.py:8]:
```suggestion
    # ANCHOR_END: prepare-sweep
```

File: src/guide/closed_channels.md [hunk 17]
```diff
@@ -0,0 +1,143 @@
...
+```dart,ignore
+{{#include ../../snippets/dart_snippets/lib/closed_channel.dart:prepare-sweep}}
+```
+</section>
+
+<div slot="title">Python</div>
+<section>
+
+```python,ignore
+{{#include ../../snippets/python/src/closed_channel.py:prepare_sweep}}
```
Comment (dangeross) [inline src/guide/closed_channels.md:54]:
```suggestion
{{#include ../../snippets/python/src/closed_channel.py:prepare-sweep}}
```

File: snippets/csharp/ClosedChannel.cs [hunk 18]
```diff
@@ -0,0 +1,43 @@
...
+{
+    public void PrepareSweep(BlockingBreezServices sdk, uint feeRate) 
+    {
+        // ANCHOR: prepare-sweep  
+        var destinationAddress = "bc1..";
+        var satPerVbyte = feeRate; 
+        try 
+        {
+            var prepareSweepResp = sdk.PrepareSweep(
+                new PrepareSweep Request(
```
Comment (dangeross) [inline snippets/csharp/ClosedChannel.cs:13]:
```suggestion
                new PrepareSweepRequest(
```

File: snippets/csharp/ClosedChannel.cs [hunk 19]
```diff
@@ -0,0 +1,43 @@
...
+                ));
+        }
+        catch (Exception)
+        {
+            // Handle error
+        }
+         // ANCHOR_END: prepare-sweep
+    }
+
+    public void PrepareSweep(BlockingBreezServices sdk, uint satPerVbyte, string toAddress)
```
Comment (dangeross) [inline snippets/csharp/ClosedChannel.cs:25]:
```suggestion
    public void Sweep(BlockingBreezServices sdk, uint satPerVbyte, string toAddress) 
```

File: snippets/csharp/ClosedChannel.cs [hunk 20]
```diff
@@ -0,0 +1,43 @@
...
+        }
+         // ANCHOR_END: prepare-sweep
+    }
+
+    public void PrepareSweep(BlockingBreezServices sdk, uint satPerVbyte, string toAddress) 
+    {
+        // ANCHOR: sweep  
+        try 
+        {
+            var prepareSweepResp = sdk.Sweep(
```
Comment (dangeross) [inline snippets/csharp/ClosedChannel.cs:30]:
```suggestion
            var sweepResp = sdk.Sweep(
```

File: snippets/csharp/ClosedChannel.cs [hunk 21]
```diff
@@ -0,0 +1,43 @@
+using Breez.Sdk;
+
+public class ClosedChannelSnippets
+{
+    public void PrepareSweep(BlockingBreezServices sdk, uint feeRate)
```
Comment (dangeross) [inline snippets/csharp/ClosedChannel.cs:5]:
```suggestion
    public void PrepareSweep(BlockingBreezServices sdk, ulong feeRate) 
```

**Review by dangeross — CHANGES_REQUESTED** [review]

File: snippets/dart_snippets/lib/closed_channel.dart [hunk 22]
```diff
@@ -0,0 +1,27 @@
+import 'package:breez_sdk/breez_sdk.dart';
+import 'package:breez_sdk/bridge_generated.dart';
+
+Future<PrepareSweepResponse> prepareSweep({required int feeRate}) async {
+  // ANCHOR: prepare-sweep
+  final satPerVbyte = feeRate;
```
Comment (erdemyerebasmaz) [inline snippets/dart_snippets/lib/closed_channel.dart:6]:
```suggestion
Future<PrepareSweepResponse> prepareSweep({required int satPerVbyte}) async {
  // ANCHOR: prepare-sweep
```

File: snippets/dart_snippets/lib/closed_channel.dart [hunk 23]
```diff
@@ -0,0 +1,27 @@
...
+  final resp = await BreezSDK().prepareSweep(req: req);
+  // ANCHOR_END: prepare-sweep
+  return resp;
+}
+
+Future<SweepResponse> sweep(
+    {required int feeRate, required String toAddress}) async {
+  // ANCHOR: sweep
+  final satPerVbyte = feeRate;
+  SweepRequest req = SweepRequest(
```
Comment (erdemyerebasmaz) [inline snippets/dart_snippets/lib/closed_channel.dart:20]: Requires reformatting, don't commit these changes using GitHub UI.

```suggestion
Future<SweepResponse> sweep(
    {required int satPerVbyte, required String toAddress,}) async {
  // ANCHOR: sweep
  SweepRequest req = SweepRequest(
```

**Review by erdemyerebasmaz — COMMENTED** [review]
Dart snippets look good.

Suggested using satPerVbyte directly as a fn parameter.

**Comment by ubbabeck** [comment]
depends on: https://github.com/breez/breez-sdk/pull/684

File: snippets/go/closed_channel.go (hunk 1 above)
Comment (ubbabeck) [reply snippets/go/closed_channel.go]: Done

File: snippets/go/closed_channel.go (hunk 2 above)
Comment (ubbabeck) [reply snippets/go/closed_channel.go]: Good point, added

File: snippets/go/closed_channel.go (hunk 4 above)
Comment (ubbabeck) [reply snippets/go/closed_channel.go]: Added

File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ClosedChannel.kt (hunk 5 above)
Comment (ubbabeck) [reply snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ClosedChannel.kt]: Fixed

File: src/guide/closed_channels.md (hunk 17 above)
Comment (ubbabeck) [reply src/guide/closed_channels.md]: Done

File: snippets/go/closed_channel.go (hunk 3 above)
Comment (ubbabeck) [reply snippets/go/closed_channel.go]: Added

File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ClosedChannel.kt (hunk 7 above)
Comment (ubbabeck) [reply snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ClosedChannel.kt]: Added

File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ClosedChannel.kt (hunk 8 above)
Comment (ubbabeck) [reply snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ClosedChannel.kt]: Added

File: snippets/python/src/closed_channel.py (hunk 9 above)
Comment (ubbabeck) [reply snippets/python/src/closed_channel.py]: Ordered

File: snippets/python/src/closed_channel.py (hunk 10 above)
Comment (ubbabeck) [reply snippets/python/src/closed_channel.py]: Done

File: snippets/react-native/closed_channel.ts (hunk 12 above)
Comment (ubbabeck) [reply snippets/react-native/closed_channel.ts]: I was not aware of this thanks. Done

File: snippets/react-native/closed_channel.ts (hunk 14 above)
Comment (ubbabeck) [reply snippets/react-native/closed_channel.ts]: Done

File: snippets/python/src/closed_channel.py (hunk 15 above)
Comment (ubbabeck) [reply snippets/python/src/closed_channel.py]: Agree, Done

File: snippets/csharp/ClosedChannel.cs (hunk 18 above)
Comment (ubbabeck) [reply snippets/csharp/ClosedChannel.cs]: Done

File: snippets/csharp/ClosedChannel.cs (hunk 19 above)
Comment (ubbabeck) [reply snippets/csharp/ClosedChannel.cs]: Done

File: snippets/csharp/ClosedChannel.cs (hunk 20 above)
Comment (ubbabeck) [reply snippets/csharp/ClosedChannel.cs]: Done

File: snippets/csharp/ClosedChannel.cs (hunk 21 above)
Comment (ubbabeck) [reply snippets/csharp/ClosedChannel.cs]: Updated to correct type UInt

File: snippets/dart_snippets/lib/closed_channel.dart (hunk 23 above)
Comment (ubbabeck) [reply snippets/dart_snippets/lib/closed_channel.dart]: Done

File: snippets/dart_snippets/lib/closed_channel.dart (hunk 22 above)
Comment (ubbabeck) [reply snippets/dart_snippets/lib/closed_channel.dart]: I did not add it here for consistency with the other examples.

File: snippets/go/closed_channel.go [hunk 24]
```diff
@@ -0,0 +1,27 @@
+package example
+
+import (
+	"log"
+
+	"github.com/breez/breez-sdk-go/breez_sdk"
+)
+
+func PrepareRedeemOnchainFunds(feeRate uint64) {
```
Comment (dangeross) [inline snippets/go/closed_channel.go:9]:
```suggestion
func PrepareRedeemOnchainFunds(feeRate uint32) {
```

File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ClosedChannel.kt [hunk 25]
```diff
@@ -0,0 +1,31 @@
...
+            // handle error
+        }
+        // ANCHOR_END: prepare-redeem-onchain-funds 
+    }
+
+    func redeemOnchainFunds(sdk: BlockingBreezServices, satPerVbyte: UInt, toAddress: String) {
+        // ANCHOR: redeem-onchain-funds 
+         try {
+            val req = RedeemOnchainFundsRequest(toAddress, satPerVbyte)
+            prepareRedeemOnchainFundsResp = sdk.redeem-onchain-funds(req)
```
Comment (dangeross) [inline snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ClosedChannel.kt:24]:
```suggestion
            val redeemOnchainFundsResp = sdk.redeemOnchainFunds(req)
```

File: snippets/python/src/closed_channel.py [hunk 26]
```diff
@@ -0,0 +1,21 @@
...
+def prepare_redeem_onchain_funds(fee_rate):
+    # ANCHOR: prepare-redeem-onchain-funds
+    destination_address = "bc1.."
+    sat_per_vbyte = fee_rate
+    try:
+        req = breez_sdk.PrepareRedeemOnchainFundsRequest(destination_address, sat_per_vbyte)
+    # ANCHOR_END: prepare-redeem-onchain-funds
+    except Exception as error:
+        print(error)
+        raise
```
Comment (dangeross) [inline snippets/python/src/closed_channel.py:12]:
```suggestion
def prepare_redeem_onchain_funds(sdk_services, fee_rate):
    # ANCHOR: prepare-redeem-onchain-funds
    destination_address = "bc1.."
    sat_per_vbyte = fee_rate
    try:
        req = breez_sdk.PrepareRedeemOnchainFundsRequest(destination_address, sat_per_vbyte)
        result = sdk_services.prepare_redeem_onchain_funds(req)
    # ANCHOR_END: prepare-redeem-onchain-funds
        return result
    except Exception as error:
        print(error)
        raise
```

File: snippets/python/src/closed_channel.py [hunk 27]
```diff
@@ -0,0 +1,21 @@
...
+
+def redeem_onchain_funds(to_address, fee_rate):
+    try:
+        # ANCHOR: redeem-onchain-funds
+        req = breez_sdk.RedeemOnchainFundsRequest(to_address, fee_rate)
+        # ANCHOR_END: redeem-onchain-funds
+    except Exception as error:
+        print(error)
+        raise
```
Comment (dangeross) [inline snippets/python/src/closed_channel.py:21]:
```suggestion
def redeem_onchain_funds(sdk_services, to_address, fee_rate):
    try:
        # ANCHOR: redeem-onchain-funds
        req = breez_sdk.RedeemOnchainFundsRequest(to_address, fee_rate)
        result = sdk_services.redeem_onchain_funds(req)
        # ANCHOR_END: redeem-onchain-funds
        return result
    except Exception as error:
        print(error)
        raise
```

File: snippets/swift/BreezSDKExamples/Sources/ClosedChannel.swift [hunk 28]
```diff
@@ -0,0 +1,25 @@
...
+//  Created by ruben on 12/12/2023.
+//
+import Foundation
+
+import BreezSDK
+
+func prepareRedeemOnchainFunds(sdk: BlockingBreezServices, feeRate: UInt64) -> PrepareRedeemOnchainFundsResponse? {
+    // ANCHOR: prepare-redeem-onchain-funds
+    let satPerVbyte = feeRate
+    let redeemOnchainFundsRequest = try? sdk.prepareRedeemOnchainFunds(req: PrepareRedeemOnchainFundsRequest(toAddress: "bc1..", satPerVbyte: satPerVbyte))
```
Comment (dangeross) [inline snippets/swift/BreezSDKExamples/Sources/ClosedChannel.swift:14]: Think the lines are too long here and hard to read on small devices
```suggestion
    let req = PrepareRedeemOnchainFundsRequest(toAddress: "bc1..", satPerVbyte: satPerVbyte)
    let redeemOnchainFundsResponse = try? sdk.prepareRedeemOnchainFunds(req: req)
```

File: snippets/swift/BreezSDKExamples/Sources/ClosedChannel.swift [hunk 29]
```diff
@@ -0,0 +1,25 @@
...
+//  ClosedChannel.swift
+//
+//
+//  Created by ruben on 12/12/2023.
+//
+import Foundation
+
+import BreezSDK
+
+func prepareRedeemOnchainFunds(sdk: BlockingBreezServices, feeRate: UInt64) -> PrepareRedeemOnchainFundsResponse? {
```
Comment (dangeross) [inline snippets/swift/BreezSDKExamples/Sources/ClosedChannel.swift:11]:
```suggestion
func prepareRedeemOnchainFunds(sdk: BlockingBreezServices, feeRate: UInt32) -> PrepareRedeemOnchainFundsResponse? {
```

File: snippets/swift/BreezSDKExamples/Sources/ClosedChannel.swift [hunk 30]
```diff
@@ -0,0 +1,25 @@
...
+    let satPerVbyte = feeRate
+    let redeemOnchainFundsRequest = try? sdk.prepareRedeemOnchainFunds(req: PrepareRedeemOnchainFundsRequest(toAddress: "bc1..", satPerVbyte: satPerVbyte))
+    // ANCHOR_END: prepare-redeem-onchain-funds
+    return redeemOnchainFundsRequest
+}
+
+func redeemOnchainFunds(sdk: BlockingBreezServices, toAddress _: String, feeRate: UInt32) -> RedeemOnchainFundsResponse? {
+    // ANCHOR: redeem-onchain-funds
+    let satPerVbyte = feeRate
+    let redeemOnchainFundsRequest = try? sdk.redeemOnchainFunds(req: RedeemOnchainFundsRequest(toAddress: "bc1..", satPerVbyte: satPerVbyte))
```
Comment (dangeross) [inline snippets/swift/BreezSDKExamples/Sources/ClosedChannel.swift:22]: Same here
```suggestion
    let req = RedeemOnchainFundsRequest(toAddress: "bc1..", satPerVbyte: satPerVbyte)
    let redeemOnchainFundsResponse = try? sdk.redeemOnchainFunds(req: req)
```

File: snippets/go/closed_channel.go [hunk 31]
```diff
@@ -0,0 +1,27 @@
...
+	// ANCHOR: prepare-redeem-onchain-funds
+	satPerVbyte := feeRate
+	destinationAddress := "bc1.."
+	req, err := sdk.PrepareRedeemOnchainFunds(breez_sdk.PrepareRedeemOnchainFundsRequest{SatPerVbyte: satPerVbyte, ToAddress: destinationAddress})
```
Comment (dangeross) [inline snippets/go/closed_channel.go:13]: Break into two lines for readability 
```suggestion
    req := breez_sdk.PrepareRedeemOnchainFundsRequest{SatPerVbyte: satPerVbyte, ToAddress: destinationAddress}
	res, err := sdk.PrepareRedeemOnchainFunds(req)
```

File: snippets/go/closed_channel.go [hunk 32]
```diff
@@ -0,0 +1,27 @@
...
+	if err != nil {
+		log.Printf("PrepareRedeemOnchainFunds error %#v", err)
+	}
+	// ANCHOR_END: prepare-redeem-onchain-funds
+}
+
+func RedeemOnchainFunds(satPerVbyte uint32, toAddress string) {
+	// ANCHOR: redeem-onchain-funds
+	req, err := sdk.RedeemOnchainFunds(breez_sdk.RedeemOnchainFundsRequest{SatPerVbyte: satPerVbyte, ToAddress: toAddress})
```
Comment (dangeross) [inline snippets/go/closed_channel.go:22]: Same here
```suggestion
    req := breez_sdk.RedeemOnchainFundsRequest{SatPerVbyte: satPerVbyte, ToAddress: toAddress}
	res, err := sdk.RedeemOnchainFunds(req)
```

File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ClosedChannel.kt [hunk 33]
```diff
@@ -0,0 +1,31 @@
...
+import breez_sdk.*
+
+class ClosedChannel {
+    func prepareRedeemOnchainFunds(sdk: BlockingBreezServices, feeRate: UInt) {
+        // ANCHOR: prepare-redeem-onchain-funds 
+         val satPerVbyte = feeRate
+         var destinationAddress = "bc1.."
+         try {
+            val req = PrepareRedeemOnchainFundsRequest(destinationAddress, satPerVbyte)
+            prepareRedeemOnchainFundsResp = sdk.prepareRedeemOnchainFunds(req)
```
Comment (dangeross) [inline snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ClosedChannel.kt:12]:
```suggestion
            val prepareRedeemOnchainFundsResp = sdk.prepareRedeemOnchainFunds(req)
```

**Review by dangeross — CHANGES_REQUESTED** [review]

File: snippets/go/closed_channel.go (hunk 24 above)
Comment (ubbabeck) [reply snippets/go/closed_channel.go]: Done

File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ClosedChannel.kt (hunk 25 above)
Comment (ubbabeck) [reply snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ClosedChannel.kt]: Done

File: snippets/python/src/closed_channel.py (hunk 26 above)
Comment (ubbabeck) [reply snippets/python/src/closed_channel.py]: Done

File: snippets/swift/BreezSDKExamples/Sources/ClosedChannel.swift (hunk 28 above)
Comment (ubbabeck) [reply snippets/swift/BreezSDKExamples/Sources/ClosedChannel.swift]: Good point, I agree.

File: snippets/swift/BreezSDKExamples/Sources/ClosedChannel.swift (hunk 29 above)
Comment (ubbabeck) [reply snippets/swift/BreezSDKExamples/Sources/ClosedChannel.swift]: Done

File: snippets/swift/BreezSDKExamples/Sources/ClosedChannel.swift (hunk 30 above)
Comment (ubbabeck) [reply snippets/swift/BreezSDKExamples/Sources/ClosedChannel.swift]: Done

File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ClosedChannel.kt (hunk 33 above)
Comment (ubbabeck) [reply snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ClosedChannel.kt]: Done, thank you for the review :D

File: snippets/go/closed_channel.go (hunk 31 above)
Comment (ubbabeck) [reply snippets/go/closed_channel.go]: Done

File: snippets/python/src/closed_channel.py (hunk 27 above)
Comment (ubbabeck) [reply snippets/python/src/closed_channel.py]: Done

File: snippets/swift/BreezSDKExamples/Sources/ClosedChannel.swift [hunk 34]
```diff
@@ -0,0 +1,33 @@
...
+    return redeemOnchainFundsRequest
+}
+
+func redeemOnchainFunds(sdk: BlockingBreezServices, toAddress _: String, feeRate: UInt32) -> RedeemOnchainFundsResponse? {
+    // ANCHOR: redeem-onchain-funds
+    let satPerVbyte = feeRate
+    let req = RedeemOnchainFundsRequest(toAddress: "bc1..", satPerVbyte: satPerVbyte)
+    let redeemOnchainFundsResponse = try? sdk.redeemOnchainFunds(req: req)
+    // ANCHOR_END: redeem-onchain-funds
+    return redeem-onchain-fundsRequest
```
Comment (dangeross) [inline snippets/swift/BreezSDKExamples/Sources/ClosedChannel.swift:32]:
```suggestion
    return redeemOnchainFundsResponse
```

File: snippets/swift/BreezSDKExamples/Sources/ClosedChannel.swift [hunk 35]
```diff
@@ -0,0 +1,33 @@
...
+    let req = PrepareRedeemOnchainFundsRequest(toAddress: "bc1..", satPerVbyte: satPerVbyte)
+    let redeemOnchainFundsResponse = try? sdk.prepareRedeemOnchainFunds(req: req)
+        result = sdk_services.prepare_redeem_onchain_funds(req)
+    # ANCHOR_END: prepare-redeem-onchain-funds
+        return result
+    except Exception as error:
+        print(error)
+        raise
+    // ANCHOR_END: prepare-redeem-onchain-funds
+    return redeemOnchainFundsRequest
```
Comment (dangeross) [inline snippets/swift/BreezSDKExamples/Sources/ClosedChannel.swift:23]:
```suggestion
    let prepareRedeemOnchainFundsResponse = try? sdk.prepareRedeemOnchainFunds(req: req)
    // ANCHOR_END: prepare-redeem-onchain-funds
    return prepareRedeemOnchainFundsResponse
```

**Review by dangeross — CHANGES_REQUESTED** [review]
Just some issues with the swift snippets left

File: snippets/swift/BreezSDKExamples/Sources/ClosedChannel.swift (hunk 34 above)
Comment (ubbabeck) [reply snippets/swift/BreezSDKExamples/Sources/ClosedChannel.swift]: Thank you, fixed :D

File: snippets/swift/BreezSDKExamples/Sources/ClosedChannel.swift (hunk 35 above)
Comment (ubbabeck) [reply snippets/swift/BreezSDKExamples/Sources/ClosedChannel.swift]: Fixed

**Review by dangeross — APPROVED** [review]
LGTM! Just need to wait for the next release to bump the versions

File: src/guide/closed_channels.md [hunk 36]
```diff
@@ -0,0 +1,143 @@
...
+
+## Prepare Redeem Onchain Funds
+
+First you may want to call prepare the transaction in order to get the transaction weight as well as the calculated fees before eventually broadcasting it.
+
+<custom-tabs category="lang">
+<div slot="title">Rust</div>
+<section>
+
+```rust, ignore
```
Comment (ok300) [inline src/guide/closed_channels.md:13]:
```suggestion
```rust,ignore
```

File: src/guide/closed_channels.md [hunk 37]
```diff
@@ -0,0 +1,143 @@
...
+
+## Redeem Onchain Funds
+
+In order to broadcast the redeem transaction do the following.
+ 
+<custom-tabs category="lang">
+<div slot="title">Rust</div>
+<section>
+
+```rust, ignore
```
Comment (ok300) [inline src/guide/closed_channels.md:83]:
```suggestion
```rust,ignore
```

**Review by ok300 — COMMENTED** [review]
Found a few NITs, otherwise LGTM

File: snippets/swift/BreezSDKExamples/Sources/SendOnchain.swift [hunk 38]
```diff
@@ -5,12 +5,12 @@
...
 //
 
-import Foundation
 import BreezSDK
+import Foundation
 
 func GetCurrentFees(sdk: BlockingBreezServices) -> ReverseSwapPairInfo? {
     //  ANCHOR: estimate-current-reverse-swap-total-fees
-    let sendAmountSat:UInt64 = 50_000
+    let sendAmountSat: UInt64 = 50000
```
[the remark this answers is not in the document]
Comment (ubbabeck) [reply snippets/swift/BreezSDKExamples/Sources/SendOnchain.swift]: Thanks I agree. Fixed

File: src/guide/closed_channels.md (hunk 37 above)
Comment (ubbabeck) [reply src/guide/closed_channels.md]: Done

File: src/guide/closed_channels.md (hunk 36 above)
Comment (ubbabeck) [reply src/guide/closed_channels.md]: Done

File: snippets/dart_snippets/lib/closed_channel.dart (hunk 22 above)
Comment (erdemyerebasmaz) [reply snippets/dart_snippets/lib/closed_channel.dart]: @ubbabeck Where are the other Dart examples that use this format? There should be a mistake.

**Comment by ok300** [comment]
Fixes https://github.com/breez/breez-sdk/issues/677

**Review by erdemyerebasmaz — APPROVED** [review]
Dart changes are looking good.

File: snippets/react-native/flake.nix
Comment (ok300) [inline snippets/react-native/flake.nix:1]: Is this `flake.nix` checked in by mistake?

If not (and it fully works), can you add a section in the README about how to build the RN snippets with nix?

Comment (ubbabeck) [reply snippets/react-native/flake.nix:1]: Hi, it was added to this pr by mistake, but I will open a separate pr for this.

File: src/guide/receive_onchain.md [hunk 39]
```diff
@@ -373,5 +373,35 @@ To calculate the fees for a channel being opened by the LSP:
 </section>
 </custom-tabs>
 
+# SwapInfo fields
```
Comment (ok300) [inline src/guide/receive_onchain.md:376]: IMO this entire section isn't needed anymore. You already described the fields in the rustdoc.

**Review by ok300 — CHANGES_REQUESTED** [review]
Had just one comment. Otherwise LGTM.

**Review by ok300 — APPROVED** [review]
LGTM.

The conflicts with `main` are the only thing left, then IMO its ready to merge.

**Review by ok300 — APPROVED** [review]

**Pull request merged**